src/Raven.Server/Documents/Subscriptions/LiveSubscriptionPerformanceCollector.cs

```diff
@@ -0,0 +1,304 @@
+    {
+        private readonly ConcurrentDictionary<string, SubscriptionAndPerformanceConnectionStatsList> _perSubscriptionConnectionStats
+            = new ConcurrentDictionary<string, SubscriptionAndPerformanceConnectionStatsList>();
+        
+        private readonly ConcurrentDictionary<string, SubscriptionAndPerformanceBatchStatsList> _perSubscriptionBatchStats
+            = new ConcurrentDictionary<string, SubscriptionAndPerformanceBatchStatsList>();
+        
+        public LiveSubscriptionPerformanceCollector(DocumentDatabase database) : base(database)
+        {
+            var initialStats = PrepareInitialPerformanceStats().ToList();
+            if (initialStats.Count > 0)
+            {
+                Stats.Enqueue(initialStats);
+            }
+            
+            Start();
+        }
+
+        // Upon collector init only: get history stats + in progress stats (to be sent to ws)
+        protected IEnumerable<SubscriptionTaskPerformanceStats> PrepareInitialPerformanceStats()
+        {
+            var results = new List<SubscriptionTaskPerformanceStats>();
+            
+            using (Database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var allSubscriptions = Database.SubscriptionStorage.GetAllSubscriptions(context, false, 0, int.MaxValue);
+                foreach (var subscription in allSubscriptions)
+                {
+                    var subscriptionName = subscription.SubscriptionName;
+                    var subscriptionId = subscription.SubscriptionId;
+                    
+                    results.Add(new SubscriptionTaskPerformanceStats
+                    {
+                        TaskName = subscriptionName,
+                        TaskId = subscriptionId
+                    });
+                    
+                    _perSubscriptionConnectionStats[subscriptionName] = new SubscriptionAndPerformanceConnectionStatsList(subscription);
+                    _perSubscriptionBatchStats[subscriptionName] = new SubscriptionAndPerformanceBatchStatsList(subscription);
+                }
+
+                foreach (var kvp in _perSubscriptionConnectionStats)
+                {
+                    var connectionPerformance = new List<SubscriptionConnectionPerformanceStats>();
+                    
+                    var connectionState = Database.SubscriptionStorage.GetSubscriptionConnection(context, kvp.Value.Handler.SubscriptionName);
+                    if (connectionState != null)
+                    {
+                        var currentConnection = connectionState.Connection;
+                        var connectionAggregators = new List<SubscriptionConnectionStatsAggregator>();
+
+                        if (currentConnection != null)
```

Comment:
don't we have duplicate detection in etl? when we adding in progress task?

Reply:
ok like this